pkg/controller/hpaenforcement_controller.go

```diff
@@ -0,0 +1,555 @@
+	}
+
+	if r.IncludedNamespaces != nil {
+		logger.V(0).Info("HPA enforcer initialized with namespace filters.", "whitelist", *r.IncludedNamespaces)
+	}
+
+	policyreco := v1alpha1.PolicyRecommendation{}
+	if err := r.Get(ctx, req.NamespacedName, &policyreco); err != nil {
+		logger.V(0).Info("Error fetching PolicyRecommendation resource.")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !isInitialized(policyreco.Status.Conditions) {
+		logger.V(0).Info("Skipping policy enforcement as the policy recommendation is not initialized.")
+		return ctrl.Result{}, nil
+	}
+
+	var workload client.Object
+	if policyreco.Spec.WorkloadMeta.Kind == "Rollout" {
+		workload = &argov1alpha1.Rollout{}
+	} else if policyreco.Spec.WorkloadMeta.Kind == "Deployment" || policyreco.Spec.WorkloadMeta.Kind == "" {
+		workload = &v1.Deployment{}
+	} else {
+		logger.V(0).Info("Skipping policy enforcement due to unrecognizable target workload meta.")
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: policyreco.Namespace,
+		Name:      policyreco.Spec.WorkloadMeta.Name,
+	}, workload); err != nil {
+		logger.V(0).Info("Skipping policy enforcement as workload can't be fetched.")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	scaledObjects := &kedaapi.ScaledObjectList{}
+	labelSelector, err := labels.Parse(fmt.Sprintf("!%s", createdByLabelKey))
+	if err != nil {
+		logger.V(0).Error(err, "Unable to parse label selector string.")
+		return ctrl.Result{}, err
+	}
+	// List only scaledObjects not created by this controller
+	if err := r.List(ctx, scaledObjects, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(scaledObjectField, workload.GetName()),
+		LabelSelector: labelSelector,
+		Namespace:     workload.GetNamespace(),
+	}); err != nil && client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+
+	var conditions []metav1.Condition
+	var statusPatch *v1alpha1.PolicyRecommendation
+	if len(scaledObjects.Items) > 0 {
+		logger.V(0).Info("ScaledObject managed by a different controller/entity already exists for this workload. Skipping.", "workload", workload, "namespace", workload.GetNamespace(), "kind", workload.GetObjectKind(), "scaledobject", scaledObjects)
+		statusPatch, conditions = CreatePolicyPatch(policyreco, conditions, v1alpha1.HPAEnforced, metav1.ConditionFalse, ScaledObjectExistsReason, ScaledObjectExistsMessage)
+		if err := r.Status().Patch(ctx, statusPatch, client.Apply, getSubresourcePatchOptions(HPAEnforcementCtrlName)); err != nil {
+			logger.Error(err, "Error updating the status of the policy reco object")
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if int32(policyreco.Spec.CurrentHPAConfiguration.Max) <= 3 || int32(policyreco.Spec.CurrentHPAConfiguration.Min) <= 3 || policyreco.Spec.CurrentHPAConfiguration.Min > policyreco.Spec.CurrentHPAConfiguration.Max {
```

Comment:
Do we want to have this 3 as minRequiredReplicas which is already in the config?